lib/functional-constraints/requiredSamples.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const _ = require('lodash');
+const jsonschema = require('jsonschema');
+
+// todo: deal with circular dep.
+const RESOURCE_ID = '/ResourceSchema';
+const RESOURCE_METHODS = ['get', 'hook', 'list', 'search', 'create'];
+
+const check = definition => {
+  if (!definition.operation || _.get(definition, 'display.hidden')) return null;
+
+  const samples = _.get(definition, 'operation.sample', {});
+  if (!Object.keys(samples).length) {
+    return new jsonschema.ValidationError(
+      'required "sample" for non-hidden operation',
+      definition,
+      definition.id
+    );
+  }
+};
+
+module.exports = (definition, mainSchema) => {
+  let definitions = [];
+
+  if (mainSchema.id === RESOURCE_ID) {
+    definitions = RESOURCE_METHODS.map(method => definition[method]).filter(
+      Boolean
+    );
+
+    // allow method definitions to inherit the sample
+    if (definition.sample) {
+      definitions.forEach(methodDefinition => {
+        if (methodDefinition.operation && !methodDefinition.operation.sample) {
+          methodDefinition.operation.sample = definition.sample;
+        }
+      });
+    }
+
+    if (!definitions.length)
```

Comment:
Could you wrap the body in `{}`? Our eslint would yell at you, but apparently it was never set up on this repo. 😁